packages/mantine/src/form/useForm.ts

```diff
@@ -1,21 +1,49 @@
 import {useForm as useMantineForm} from '@mantine/form';
-import {GetInputProps} from '@mantine/form/lib/types';
+import {LooseKeys, UseFormInput, _TransformValues} from '@mantine/form/lib/types';
 
-export const useForm: typeof useMantineForm = (options) => {
+type InputTypes = 'input' | 'checkbox' | 'collection';
+
+type GetInputProps<Values> = <Field extends LooseKeys<Values>>(
+    path: Field,
+    options?: {type?: InputTypes; withError?: boolean; withFocus?: boolean},
+) => {
+    value: any;
+    onChange: any;
+    checked?: any;
+    error?: any;
+    onFocus?: any;
+    onBlur?: any;
+    onReorderItem?: any;
+    onRemoveItem?: any;
+    onInsertItem?: any;
+};
+
+export const useForm = <
+    Values = Record<string, unknown>,
+    TransformValues extends _TransformValues<Values> = (values: Values) => Values,
+>(
+    options: UseFormInput<Values, TransformValues> = {},
+) => {
     const form = useMantineForm(options);
 
     const getInputProps: GetInputProps<Record<string, unknown>> = (
         path,
         {type = 'input', withError = type === 'input', withFocus = true} = {},
     ) => {
-        const originalPayload = form.getInputProps(path, {type, withError, withFocus});
+        const originalPayload = form.getInputProps(path, {
+            type: type === 'collection' || 'input' ? 'input' : 'checkbox',
+            withError,
+            withFocus,
+        });
         if (Array.isArray(originalPayload.value)) {
```

Comment:
We could do the check for the type here

```suggestion
        if (type === 'collection') {
```

Reply:
Sure, and returning only ```originalPayload``` if not ?

Reply:
exactly